### DISCUSSION TITLE
Unable to install perspective-python on Mac OS 10.14

### DISCUSSION THREAD

**Discussion by SumithRZT** [perspective-dev/perspective#1430] [Q&A]
Unable to install perspective-python on Mac OS 10.14, Python 3.7.5

I just ran sudo pip3 install perspective-python as it was given in [](https://perspective.finos.org/docs/md/python.html#installation)

The installation failed with following error

 ERROR: Command errored out with exit status 1:
   command: /Library/Frameworks/Python.framework/Versions/3.7/bin/python3.7 /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /tmp/tmpjj2ds4fe
       cwd: /private/tmp/pip-install-wzaz_21e/perspective-python_cf3e6090e1e04b40b3c090218e9fd496
  Complete output (219 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.9-x86_64-3.7
  creating build/lib.macosx-10.9-x86_64-3.7/perspective
  copying perspective/libpsp.py -> build/lib.macosx-10.9-x86_64-3.7/perspective
  copying perspective/__init__.py -> build/lib.macosx-10.9-x86_64-3.7/perspective
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/core
  copying perspective/core/aggregate.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/core
  copying perspective/core/exception.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/core
  copying perspective/core/_version.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/core
  copying perspective/core/sort.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/core
  copying perspective/core/__init__.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/core
  copying perspective/core/plugin.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/core
  copying perspective/core/filters.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/core
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/widget
  copying perspective/widget/widget.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/widget
  copying perspective/widget/__init__.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/widget
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/tornado_handler
  copying perspective/tornado_handler/tornado_client.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tornado_handler
  copying perspective/tornado_handler/__init__.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tornado_handler
  copying perspective/tornado_handler/tornado_handler.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tornado_handler
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/tests
  copying perspective/tests/conftest.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests
  copying perspective/tests/__init__.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests
  copying perspective/tests/common.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/manager
  copying perspective/manager/manager_internal.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/manager
  copying perspective/manager/session.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/manager
  copying perspective/manager/__init__.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/manager
  copying perspective/manager/manager.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/manager
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/viewer
  copying perspective/viewer/viewer_traitlets.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/viewer
  copying perspective/viewer/viewer.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/viewer
  copying perspective/viewer/__init__.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/viewer
  copying perspective/viewer/validate.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/viewer
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/table
  copying perspective/table/_constants.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/table
  copying perspective/table/_accessor.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/table
  copying perspective/table/__init__.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/table
  copying perspective/table/_data_formatter.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/table
  copying perspective/table/view_config.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/table
  copying perspective/table/_callback_cache.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/table
  copying perspective/table/view.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/table
  copying perspective/table/_state.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/table
  copying perspective/table/_date_validator.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/table
  copying perspective/table/table.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/table
  copying perspective/table/_utils.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/table
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/client
  copying perspective/client/dispatch.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/client
  copying perspective/client/client.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/client
  copying perspective/client/__init__.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/client
  copying perspective/client/table_api.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/client
  copying perspective/client/view_api.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/client
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/core/data
  copying perspective/core/data/__init__.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/core/data
  copying perspective/core/data/np.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/core/data
  copying perspective/core/data/pd.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/core/data
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/tests/core
  copying perspective/tests/core/test_plugin.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/core
  copying perspective/tests/core/test_async.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/core
  copying perspective/tests/core/__init__.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/core
  copying perspective/tests/core/test_aggregates.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/core
  copying perspective/tests/core/test_sort.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/core
  copying perspective/tests/core/test_threadpool.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/core
  copying perspective/tests/core/test_layout.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/core
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/tests/widget
  copying perspective/tests/widget/__init__.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/widget
  copying perspective/tests/widget/test_widget.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/widget
  copying perspective/tests/widget/test_widget_pandas.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/widget
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/tests/tornado_handler
  copying perspective/tests/tornado_handler/test_tornado_handler_chunked.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/tornado_handler
  copying perspective/tests/tornado_handler/__init__.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/tornado_handler
  copying perspective/tests/tornado_handler/test_tornado_handler.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/tornado_handler
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/tests/manager
  copying perspective/tests/manager/test_session.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/manager
  copying perspective/tests/manager/__init__.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/manager
  copying perspective/tests/manager/test_manager.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/manager
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/tests/viewer
  copying perspective/tests/viewer/__init__.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/viewer
  copying perspective/tests/viewer/test_viewer.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/viewer
  copying perspective/tests/viewer/test_validate.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/viewer
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_table_datetime.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_table.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_view_computed.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/__init__.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/object_sequence.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_table_infer.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_table_object.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_delete.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_table_pandas.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_update.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_table_numpy.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_view.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_update_numpy.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_update_arrow.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_to_format.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_exception.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_table_arrow.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_update_pandas.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_ports.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_remove.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  copying perspective/tests/table/test_to_arrow.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  running egg_info
  writing perspective_python.egg-info/PKG-INFO
  writing dependency_links to perspective_python.egg-info/dependency_links.txt
  writing requirements to perspective_python.egg-info/requires.txt
  writing top-level names to perspective_python.egg-info/top_level.txt
  reading manifest file 'perspective_python.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  warning: no files found matching '.bumpversion.cfg'
  warning: no files found matching 'Makefile'
  warning: no files found matching 'perspective/table/*.so'
  warning: no files found matching 'perspective/table/*.dll'
  warning: no files found matching 'perspective/table/*.dylib'
  warning: no previously-included files found matching 'docs/#*'
  no previously-included directories found matching 'docs/build'
  no previously-included directories found matching 'docs/gh-pages'
  no previously-included directories found matching 'docs/dist'
  no previously-included directories found matching 'bench'
  warning: no directories found matching 'dist/test'
  warning: no previously-included files matching '*~' found anywhere in distribution
  warning: no previously-included files matching '*.pyc' found anywhere in distribution
  warning: no previously-included files matching '*.pyo' found anywhere in distribution
  warning: no previously-included files matching '.git' found anywhere in distribution
  warning: no previously-included files matching '.ipynb_checkpoints' found anywhere in distribution
  warning: no previously-included files matching '.DS_Store' found anywhere in distribution
  writing manifest file 'perspective_python.egg-info/SOURCES.txt'
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/include
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/include/perspective
  copying perspective/include/perspective/python.h -> build/lib.macosx-10.9-x86_64-3.7/perspective/include/perspective
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/include/perspective/python
  copying perspective/include/perspective/python/accessor.h -> build/lib.macosx-10.9-x86_64-3.7/perspective/include/perspective/python
  copying perspective/include/perspective/python/base.h -> build/lib.macosx-10.9-x86_64-3.7/perspective/include/perspective/python
  copying perspective/include/perspective/python/column.h -> build/lib.macosx-10.9-x86_64-3.7/perspective/include/perspective/python
  copying perspective/include/perspective/python/computed.h -> build/lib.macosx-10.9-x86_64-3.7/perspective/include/perspective/python
  copying perspective/include/perspective/python/context.h -> build/lib.macosx-10.9-x86_64-3.7/perspective/include/perspective/python
  copying perspective/include/perspective/python/fill.h -> build/lib.macosx-10.9-x86_64-3.7/perspective/include/perspective/python
  copying perspective/include/perspective/python/numpy.h -> build/lib.macosx-10.9-x86_64-3.7/perspective/include/perspective/python
  copying perspective/include/perspective/python/serialization.h -> build/lib.macosx-10.9-x86_64-3.7/perspective/include/perspective/python
  copying perspective/include/perspective/python/table.h -> build/lib.macosx-10.9-x86_64-3.7/perspective/include/perspective/python
  copying perspective/include/perspective/python/utils.h -> build/lib.macosx-10.9-x86_64-3.7/perspective/include/perspective/python
  copying perspective/include/perspective/python/view.h -> build/lib.macosx-10.9-x86_64-3.7/perspective/include/perspective/python
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/src
  copying perspective/src/accessor.cpp -> build/lib.macosx-10.9-x86_64-3.7/perspective/src
  copying perspective/src/column.cpp -> build/lib.macosx-10.9-x86_64-3.7/perspective/src
  copying perspective/src/computed.cpp -> build/lib.macosx-10.9-x86_64-3.7/perspective/src
  copying perspective/src/context.cpp -> build/lib.macosx-10.9-x86_64-3.7/perspective/src
  copying perspective/src/fill.cpp -> build/lib.macosx-10.9-x86_64-3.7/perspective/src
  copying perspective/src/numpy.cpp -> build/lib.macosx-10.9-x86_64-3.7/perspective/src
  copying perspective/src/python.cpp -> build/lib.macosx-10.9-x86_64-3.7/perspective/src
  copying perspective/src/serialization.cpp -> build/lib.macosx-10.9-x86_64-3.7/perspective/src
  copying perspective/src/table.cpp -> build/lib.macosx-10.9-x86_64-3.7/perspective/src
  copying perspective/src/utils.cpp -> build/lib.macosx-10.9-x86_64-3.7/perspective/src
  copying perspective/src/view.cpp -> build/lib.macosx-10.9-x86_64-3.7/perspective/src
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/tests/client_mode
  copying perspective/tests/client_mode/test_client_mode.py -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/client_mode
  copying perspective/tests/table/psp_test -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table
  creating build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table/arrow
  copying perspective/tests/table/arrow/date32.arrow -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table/arrow
  copying perspective/tests/table/arrow/date64.arrow -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table/arrow
  copying perspective/tests/table/arrow/dict.arrow -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table/arrow
  copying perspective/tests/table/arrow/dict_update.arrow -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table/arrow
  copying perspective/tests/table/arrow/int_float_str.arrow -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table/arrow
  copying perspective/tests/table/arrow/int_float_str_file.arrow -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table/arrow
  copying perspective/tests/table/arrow/int_float_str_update.arrow -> build/lib.macosx-10.9-x86_64-3.7/perspective/tests/table/arrow
  running build_ext
  Traceback (most recent call last):
    File "/Library/Frameworks/Python.framework/Versions/3.7/bin/cmake", line 6, in <module>
      from cmake import cmake
  ModuleNotFoundError: No module named 'cmake'
  Traceback (most recent call last):
    File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 280, in <module>
      main()
    File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 263, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 205, in build_wheel
      metadata_directory)
    File "/private/tmp/pip-build-env-0kdojr3_/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 222, in build_wheel
      wheel_directory, config_settings)
    File "/private/tmp/pip-build-env-0kdojr3_/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 207, in _build_with_temp_dir
      self.run_setup()
    File "/private/tmp/pip-build-env-0kdojr3_/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 259, in run_setup
      self).run_setup(setup_script=setup_script)
    File "/private/tmp/pip-build-env-0kdojr3_/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 150, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 264, in <module>
      cmdclass=dict(build_ext=PSPBuild, sdist=PSPCheckSDist),
    File "/private/tmp/pip-build-env-0kdojr3_/overlay/lib/python3.7/site-packages/setuptools/__init__.py", line 153, in setup
      return distutils.core.setup(**attrs)
    File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/dist.py", line 966, in run_commands
      self.run_command(cmd)
    File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/private/tmp/pip-build-env-0kdojr3_/overlay/lib/python3.7/site-packages/wheel/bdist_wheel.py", line 299, in run
      self.run_command('build')
    File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/command/build.py", line 135, in run
      self.run_command(cmd_name)
    File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "setup.py", line 109, in run
      self.run_cmake()
    File "setup.py", line 114, in run_cmake
      out = subprocess.check_output([self.cmake_cmd, "--version"])
    File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 411, in check_output
      **kwargs).stdout
    File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 512, in run
      output=stdout, stderr=stderr)
  subprocess.CalledProcessError: Command '['/Library/Frameworks/Python.framework/Versions/3.7/bin/cmake', '--version']' returned non-zero exit status 1.
  ----------------------------------------
  ERROR: Failed building wheel for perspective-python
Failed to build perspective-python
ERROR: Could not build wheels for perspective-python which use PEP 517 and cannot be installed directly

**Comment by sc1f** [comment]
Please run `pip install -v` for verbose output - the default output does not have the information we need for debugging.

**Reply by SumithRZT** [reply]
[install_logs.txt](https://github.com/finos/perspective/files/6533846/install_logs.txt)
Please find the logs here

**Reply by sc1f** [reply]
You need to install CMake - Homebrew should work.

**Reply by SumithRZT** [reply]
cmake is already installed the version is  _cmake version 3.20.2_

**Reply by timkpaine** [reply]
You installed with sudo, is it available to root? `sudo which cmake`

**Reply by SumithRZT** [reply]
cmake is also installed as pip package. Still the same issue

**Reply by SumithRZT** [reply]
`sudo which cmake` is pointing to` /Library/Frameworks/Python.framework/Versions/3.7/bin/cmake`

**Reply by sc1f** [reply]
From the logs:

```python
2021-05-24T22:29:12,772     File "/Library/Frameworks/Python.framework/Versions/3.7/bin/cmake", line 6, in <module>
2021-05-24T22:29:12,772       from cmake import cmake
2021-05-24T22:29:12,772   ModuleNotFoundError: No module named 'cmake'
```
I am not sure exactly what is going on with this error,  but I would uninstall the CMake that was installed with pip, and re-install CMake using Homebrew. You should be able to verify that CMake is on your PATH and valid by running `cmake --version` or `brew list cmake` afterwards.

**Reply by SumithRZT** [reply]
Ok let me try that

**Reply by SumithRZT** [reply]
I uninstalled pip cmake and installed with brew,  `brew list cmake` is working now.  but when i try to do `sudo pip3 install perspective-python ` I am getting these new errors
[ppython.txt](https://github.com/finos/perspective/files/6539893/ppython.txt)

**Reply by sc1f** [reply]
Since we no longer build wheels for OSX below 10.15, you can use the [developer docs](https://perspective.finos.org/docs/md/development.html#macososx) and install the listed dependencies using Homebrew.

From your logs:

```bash
2021-05-25T19:35:36,055     /tmp/pip-install-o9hntyq0/perspective-python_511c466747534ebeaf1e509eaeb5d33c/dist/cmake/modules/FindFlatbuffers.cmake:67 (find_package_handle_standard_args)
2021-05-25T19:35:36,055     /tmp/pip-install-o9hntyq0/perspective-python_511c466747534ebeaf1e509eaeb5d33c/build/temp.macosx-10.9-x86_64-3.7/arrow-src/cpp/CMakeLists.txt:177 (find_package)
2021-05-25T19:35:36,055   This warning is for project developers.  Use -Wno-dev to suppress it.

2021-05-25T19:35:36,055   -- Could NOT find FLATBUFFERS (missing: FLATBUFFERS_COMPILER)
```

**Reply by SumithRZT** [reply]
If I have a ubuntu machine how this installation works? I want to know if there are any dependencies has to be taken care of like cmake

**Reply by sc1f** [reply]
Our wheels should work for Linux - otherwise docs: https://perspective.finos.org/docs/md/development.html#system-dependencies

**Reply by SumithRZT** [reply]
After the installation of flatbuffers it is working. But is pagination available for data grids ?. It's taking a lot of time to create a perspective table for 30 mb csv file

**Reply by texodus** [reply]
Pagination will not make loading a CSV faster.  Use Apache Arrow instead of CSV if you want exceptional load performance.  You can do this trivially with `View.to_arrow()` in  `perspective-python`.

**Reply by SumithRZT** [reply]
Loading a CSV is Quick but I am passing that dataframe to convert it into a Perspective Table, that part is taking a lot of time
